clients/vscode-hlasmplugin/src/hlasmExternalFiles.ts

```diff
@@ -341,12 +353,76 @@ export class HLASMExternalFiles {
 
     }
 
+    private deriveCacheEntryName(clientId: string, service: string, normalizedPath: string) {
+        return cacheVersion + '.' + crypto.createHash('sha256').update(JSON.stringify([
+            clientId,
+            service,
+            normalizedPath
+        ])).digest().toString('hex');
+    }
+
+    private async getCachedResult(client: ExternalFilesClient, service: string, normalizedPath: string, expect: 'string'): Promise<string | typeof not_exists | undefined>;
+    private async getCachedResult(client: ExternalFilesClient, service: string, normalizedPath: string, expect: 'arrayOfStrings'): Promise<string[] | typeof not_exists | undefined>;
+    private async getCachedResult(client: ExternalFilesClient, service: string, normalizedPath: string, expect: 'string' | 'arrayOfStrings') {
+        if (!this.cache) return undefined;
+        const clientId = await client.uniqueId();
+        if (!clientId) return undefined;
+
+        const cacheEntryName = vscode.Uri.joinPath(this.cache.uri, this.deriveCacheEntryName(clientId, service, normalizedPath));
+
+        try {
+            const cachedResult = JSON.parse(new TextDecoder().decode(await promisify(inflate)(await this.cache.fs.readFile(cacheEntryName))));
+
+            if (cachedResult === 'not_exists') return not_exists;
+            if (cachedResult instanceof Object && 'data' in cachedResult) {
+                const data = cachedResult.data;
+
+                if (expect === 'string' && typeof data === 'string')
+                    return data;
+                if (expect === 'arrayOfStrings' && Array.isArray(data) && data.every(x => typeof x === 'string'))
+                    return data;
+            }
+        }
+        catch (e) { }
+
+        return undefined;
+    }
+
+    private async storeCachedResult(client: ExternalFilesClient, service: string, normalizedPath: string, value: string | string[] | typeof not_exists) {
+        if (!this.cache) return;
+        const clientId = await client.uniqueId();
+        if (!clientId) return;
+
+        const cacheEntryName = vscode.Uri.joinPath(this.cache.uri, this.deriveCacheEntryName(clientId, service, normalizedPath));
+
+        try {
+            const data = value === not_exists
+                ? new TextEncoder().encode(JSON.stringify("not_exists"))
```

Comment:
Maybe turn this into a constant?

Reply:
meaning "not_exists"

Reply:
I'll probably open another PR with extra changes soon, so I'll include it there.